src/Paginator.php

```diff
@@ -865,7 +865,7 @@ protected function _getCacheInternalId()
         // @codingStandardsIgnoreEnd
         return md5(
             get_class($this->getAdapter())
-            . json_encode($this->getAdapter())
+            . hash('sha512', print_r($this->getAdapter(), true))
```

Comment:
Wouldn't `spl_object_hash($this->getAdapter())` make more sense here?

Reply:
@weierophinney 
See your own comment: https://github.com/zendframework/zend-paginator/issues/41#issuecomment-341519318

Reply:
`spl_object_hash` has been used in the past: https://github.com/zendframework/zend-paginator/commit/54f4f4c1f05bfe7f581657604998d492ac7af29c#diff-b7dae88f55a82631386bada74d3d1e9fL863

Reply:
Ah, okay, that makes sense again. (I'm getting old and can't remember this stuff!)

Still need unit tests, though!